boa/src/value/mod.rs

```diff
@@ -664,7 +664,11 @@ impl Value {
                     ObjectData::String(string.clone()),
                 ));
                 // Make sure the correct length is set on our new string object
-                object.set("length".into(), string.chars().count().into());
+                object.insert_property(
+                    PropertyKey::String("length".into()),
+                    Value::from(string.chars().count()),
```

Comment:
Just now noticed, this doesn't seem to match the `length` calculation in `boa/src/builtins/string/mod.rs`, there we use `encode_utf16().count()`, so I assume this one might give different results. I'd like someone else to back me on this though.

Reply:
@Razican can you confirm this?

Reply:
Seems that `chars().count()` counts the code points, and `encode_utf16().count()` counts the UTF-16 code units, and I think that counting UTF-16 code units is what JavaScript does, but I am not quite sure.

Reply:
As per [this](https://tc39.es/ecma262/#sec-terms-and-definitions-string-value), the inner value of a String is an array of `u16` code points (which we currently don't use, but at some point we should).

Then, as per [this](https://tc39.es/ecma262/#sec-properties-of-string-instances-length), the `length` property is the number of values (`u16` code points). So it should use `encode_utf16().count()`.

Reply:
That's what I thought, thanks. Just change the method invoked and it should be good to `merge` @tofpie